src/controllers/eventController.js

```diff
@@ -65,6 +67,34 @@ eventController.post('/register', async (req, res) => {
     }
 });
 
+// Upcoming Events
+eventController.get('/upcomingEvents', async (req, res) => {
+    try {
+        const { todayStart } = getNeededDates();
+        const events = await getUpcomingEvents({ todayStart });
+
+        res.status(200).json(events);
+        res.end();
+    } catch (error) {
+        res.status(400).json(error.message);
+        res.end();
+    }
+});
+
+// Past events
+eventController.get('/pastEvents', async (req, res) => {
+    try {
+        const { todayStart } = getNeededDates();
```

Comment:
You don't need it like this. Now you're calculating "startDate" and "endDate" from this function without needing them. Just put and calculate only what you need.  Even if this means to separate it in a new function.